app/Domain/Dice/Dice.cpp

```diff
@@ -0,0 +1,53 @@
+//
+// Created by Gabriel Noriega on 2018-02-14.
+//
+
+#include <random>
+#include "Dice.h"
+#include <iomanip>
+
+using std::cout;
+using std::endl;
+
+
+int Dice::generateRandomNumber() {
+    const int range_from = 0;
+    const int range_to = 6;
+    std::random_device rand_dev;
+    std::mt19937 generator(rand_dev());
+    std::uniform_int_distribution<int> distr(range_from, range_to);
+    return distr(generator);
+}
+
+int Dice::transformRoll(int value) const {
+    (value < 3) ? (value = 0) : (value -= 3);
```

Comment:
Suggested to refactor as `value = (value < 3) ? value - value : value - 3;`

Reply:
Shouldn't you simply reduce the range to [0,3] rather than adding an extra op to map [0,6] to [0,3]?

Reply:
Well, really they both do the same thing, no need for the value - value. I'll change it if you think  it is more readable. I'll explain why the range in a following comment.

Reply:
In order to keep a Dice as real as possible, I purposely made it this way. A die has 6 faces, 3 of which are blank, and the other are the numbers 1, 2, and 3. The way I implement it will simulate a die as close as possible as a real one.

Reply:
I do think it's more readable; you can directly do `return (value < 3) ? value - value : value - 3`. Better still you can do `return (value < 3) ? 0 : value - 3`;

Reply:
Oh damn, yeah I was actually looking for that. I don't know why I didn't think of that. Great, thanks.